scripts/zones/Arrapago_Reef/mobs/Medusa.lua

```diff
@@ -1,28 +1,45 @@
 -----------------------------------
 -- Area: Arrapago Reef
 -- NPC:  Medusa
+-- TODO: resists, attack/def boosts
 -----------------------------------
 
 require("scripts/globals/titles");
+require("scripts/zones/Arrapago_Reef/MobIDs");
+require("scripts/zones/Arrapago_Reef/TextIDs");
 
 -----------------------------------
 -- onMobSpawn Action
 -----------------------------------
 
 function onMobSpawn(mob)
+    mob:setLocalVar("eeshpp", math.random(5,99)); -- Uses EES randomly during the fight
 end;
 
 -----------------------------------
 -- onMobEngaged Action
 -----------------------------------
 
-function onMobEngaged(mob,target)
+function onMobEngaged(mob, target)
+    target:showText(mob, MEDUSA_ENGAGE);
+    SpawnMob(ExonOne,   180):updateEnmity(target);
+    SpawnMob(ExonTwo,   180):updateEnmity(target);
+    SpawnMob(ExonThree, 180):updateEnmity(target);
+    SpawnMob(ExonFour,  180):updateEnmity(target);
+end;
 
-	SpawnMob(16998863,180):updateEnmity(target);
-	SpawnMob(16998864,180):updateEnmity(target);
-	SpawnMob(16998865,180):updateEnmity(target);
-	SpawnMob(16998866,180):updateEnmity(target);
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
 
+function onMobFight(mob, target)
+    local HPP = mob:getHPP();
+    if (mob:getLocalVar("usedees") == 0) then
+        if (HPP <= mob:getLocalVar("eeshpp")) then
+            mob:useMobAbility(479); -- Eagle Eye Shot

```

Comment:
No, no they do NOT change. The skill ID never, ever gets changes - the different ID is a totally different skill. The localVar is handling tracking, and gets wiped out when no longer needed on its own. It being handle din core is precisely what I have an issue with. You'll wind up with a  bunch of redundant hard coded checks in core to see which version of Eagle Eye Shot to use, instead of just calling the correct one form that specific mobs script. How is that better?